portal-api/api-tests/v1/banks/banks.api-test.js

```diff
@@ -1,176 +1,172 @@
 const { ADMIN } = require('../../../src/v1/roles/roles');
 
 const aBank = require('./bank-builder');
-const { DB_COLLECTIONS } = require('../../fixtures/constants');
+const { withClientAuthenticationTests } = require('../../common-tests/client-authentication-tests');
+const { withNoRoleAuthorisationTests, withRoleAuthorisationTests } = require('../../common-tests/role-authorisation-tests');
+
+const newBanks = ['112233', '112234', '112235'].map((id) => aBank({ id, name: `Bank${id}` }));
 
-const newBank = aBank({ id: '112233' });
 const updatedBank = aBank({
-  id: '112233',
+  id: newBanks[0].id,
   bankName: 'Updated bank name',
 });
 
-// Disabled MGA tests as they remove banks from DB which other functionality now depends on
-// so other test will break if this is run before
-describe.skip('/v1/banks', () => {
+describe('/v1/banks', () => {
   let anAdmin;
-  let aNonAdmin;
+  let testUsers;
+  let noRoles;
 
   beforeAll(async () => {
-    const testUsers = await testUserCache.initialise(app);
+    testUsers = await testUserCache.initialise(app);
     anAdmin = testUsers().withRole(ADMIN).one();
-    aNonAdmin = testUsers().withoutRole(ADMIN).one();
+    noRoles = testUsers().withoutAnyRoles().one();
   });
 
   beforeEach(async () => {
-    await databaseHelper.wipe([DB_COLLECTIONS.BANKS]);
+    await databaseHelper.deleteBanks(newBanks.map(({ id }) => id));
   });
 
   describe('GET /v1/banks', () => {
-    it('rejects requests that do not present a valid Authorization token', async () => {
-      const { status } = await as().get('/v1/banks');
-
-      expect(status).toEqual(401);
+    withClientAuthenticationTests({
+      makeRequestWithoutAuthHeader: () => get('/v1/banks'),
+      makeRequestWithAuthHeader: (authHeader) => get('/v1/banks', { headers: { Authorization: authHeader } }),
     });
 
-    it('accepts requests that present a valid Authorization token', async () => {
-      const { status } = await as(aNonAdmin).get('/v1/banks');
-
-      expect(status).toEqual(200);
+    withNoRoleAuthorisationTests({
+      getUserWithRole: (role) => testUsers().withRole(role).one(),
+      getUserWithoutAnyRoles: () => noRoles,
+      makeRequestAsUser: (user) => as(user).get('/v1/banks'),
+      successStatusCode: 200,
     });
 
     it('returns a list of banks', async () => {
-      const banks = [aBank({ id: '1' }), aBank({ id: '2' }), aBank({ id: '3' })];
-
-      await as(anAdmin).postEach(banks).to('/v1/banks');
+      await as(anAdmin).postEach(newBanks).to('/v1/banks');
 
-      const { status, body } = await as(aNonAdmin).get('/v1/banks');
+      const { status, body } = await as(anAdmin).get('/v1/banks');
 
+      const expectedBanks = expectMongoIds(newBanks.map((bank) => ({ ...bank, auditRecord: expectAnyPortalUserAuditDatabaseRecord() })));
       expect(status).toEqual(200);
-      expect(body.banks).toEqual(expectMongoIds(banks));
+      expect(body.banks).toEqual(expect.arrayContaining(expectedBanks));
```

Comment:
This Array contains all MDL banks too, but this test was skipped before because we can't delete every bank without breaking some tests